File: backend/app/schemas/testimonial.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field

class TestimonialBase(BaseModel):
    name: str
    designation: Optional[str] = ""
    company: Optional[str] = ""
    content: str
    rating: int = 5
    photo: Optional[str] = ""
    enabled: bool = True
    featured: bool = False
    display_order: int = 1
    status: str = "published"  # draft, published, archived

class TestimonialCreate(TestimonialBase):
    pass

class TestimonialUpdate(BaseModel):
    name: Optional[str] = None
    designation: Optional[str] = None
    company: Optional[str] = None
    content: Optional[str] = None
    rating: Optional[int] = None
    photo: Optional[str] = None
    enabled: Optional[bool] = None
    featured: Optional[bool] = None
    display_order: Optional[int] = None
    status: Optional[str] = None

class TestimonialResponse(TestimonialBase):
    id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

File: admin/lib/types/testimonial.ts
import { ContentStatus } from "./product";

export interface Testimonial {
  id: string;
  name: string;
  designation?: string;
  company?: string;
  content: string;
  rating: number; // 1-5
  photo_url?: string;
  featured: boolean;
  enabled: boolean;
  verified?: boolean;
  display_order: number;
  status: ContentStatus;
  created_at?: string;
  updated_at?: string;
}
